src/main/java/ch/naviqore/app/controller/GlobalValidator.java

```diff
@@ -0,0 +1,54 @@
+package ch.naviqore.app.controller;
+
+import ch.naviqore.service.ScheduleInformationService;
+import ch.naviqore.service.Stop;
+import ch.naviqore.service.exception.StopNotFoundException;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+@NoArgsConstructor(access = AccessLevel.NONE)
+final class GlobalValidator {
+
+    public static LocalDateTime validateAndSetDefaultDateTime(LocalDateTime dateTime,
+                                                              ScheduleInformationService service) {
+        dateTime = (dateTime == null) ? LocalDateTime.now() : dateTime;
+
+        if (service.getValidity().isWithin(dateTime.toLocalDate())) {
+            return dateTime;
+        }
+
+        LocalDateTime startDate = service.getValidity().getStartDate().atStartOfDay();
```

Comment:
Just checked the multi day raptor implementation. This should handle days correctly when one or more days are with no active calendar.

Which made me think about the validity implementation again. Personally I would calculate the smallest and largest date from all calendars and exceptions once (probably within the schedule, as this will be reinitialized when the schedule updates) and refer to those references when using the validity object. isBetween() should really just check if the date is between those two dates, because even if there's no connection on the given day and mutli day scanning is enabled, potentially the next day could be active and return connections.

Reply:
Another inconsistency I just realized is that GtfsSchedule.getNextDepartures() does only scan for the active service day. So maybe we should include the until argument (default = null) and add at least some multi day logic so when the limit of the day is not hit, the next day is included and so on until the limit is reached (or a certain number of days is exceeded)